### ISSUE TITLE
Lab 01 - docker file seems incorrect - fix included

### ISSUE THREAD

**Issue by jeanpaulsmit** [MicrosoftLearning/AZ-203-DevelopingSolutionsforMicrosoftAzure#19]
# Module: 01
#### Step: 3

First of all, thanks for this awesome learning material!

However the Dockerfile seems to be incorrect.
Two things are incorrect in my view:
1)  the code is .NET Core code which needs to run on a .NET Core image to be able to run on a Linux image but the image in the Dockerfile points to the .NET framework. 
2) the .NET framework image could not be found

Below the Dockerfile I used in the lab:

FROM mcr.microsoft.com/dotnet/core/sdk:2.2 AS build
WORKDIR /app

COPY *.csproj ./
RUN dotnet restore

COPY . ./
RUN dotnet publish --configuration Release --output out --runtime ubuntu.16.04-x64 --self-contained

FROM mcr.microsoft.com/dotnet/core/runtime:2.2
WORKDIR /app
COPY --from=build /app/out .

**Comment by seesharprun** [comment]
The Dockerfile is based off the old images that were natively hosted on Docker Hub.

Since then Microsoft released their own container registry and changed the name of all containers.

https://azure.microsoft.com/en-us/blog/microsoft-syndicates-container-catalog/

A fix will land shortly.

**Comment by SychevIgor** [comment]
Will be nice to have this fix soon. I'm delivering this training today.

**Comment by seesharprun** [comment]
I updated the Dockerfile based on the new [mcr.microsoft.com](https://azure.microsoft.com/en-us/blog/microsoft-syndicates-container-catalog/) container registry. As part of updating the Dockerfile, this fix was no longer needed.

The fixed Dockerfile can be found in [Version 20190701.06](https://github.com/MicrosoftLearning/AZ-203-DevelopingSolutionsforMicrosoftAzure/releases/tag/v20190701.06).

Here is the contents of the corrected Dockerfile:

```
FROM mcr.microsoft.com/dotnet/core/sdk:2.2-alpine AS build
WORKDIR /app

COPY *.csproj ./
RUN dotnet restore

COPY . ./
RUN dotnet publish --configuration Release --output out

FROM mcr.microsoft.com/dotnet/core/runtime:2.2-alpine
WORKDIR /app

COPY --from=build /app/out .

ENTRYPOINT ["dotnet", "ipcheck.dll"]
```
